chat-android/src/main/java/com/ably/chat/ChatApi.kt

```diff
@@ -0,0 +1,149 @@
+package com.ably.chat
+
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import io.ably.lib.http.HttpCore
+import io.ably.lib.http.HttpUtils
+import io.ably.lib.types.AblyException
+import io.ably.lib.types.AsyncHttpPaginatedResponse
+import io.ably.lib.types.ErrorInfo
+import io.ably.lib.types.Param
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+private const val API_PROTOCOL_VERSION = 3
+private const val PROTOCOL_VERSION_PARAM_NAME = "v"
+private val apiProtocolParam = Param(PROTOCOL_VERSION_PARAM_NAME, API_PROTOCOL_VERSION.toString())
+
+// TODO make this class internal
+class ChatApi(private val realtimeClient: RealtimeClient) {
+
+    suspend fun getMessages(roomId: String, params: QueryOptions): PaginatedResult<Message> {
+        return makeAuthorizedPaginatedRequest(
+            url = "/chat/v1/rooms/$roomId/messages",
+            method = "GET",
+            params = params.toParams(),
+        ) {
+            Message(
+                timeserial = it.asJsonObject.get("timeserial").asString,
```

Comment:
I added explicit validation and couple of unit tests that check we are silently ignore stuff we aren't expecting and  if response is invalid or required fields are missing it will throw AblyException